components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -118,6 +119,9 @@ open class GleanInternalAPI internal constructor () {
         // This must be set before anything that might trigger the sending of pings.
         initialized = true
 
+        // Signal Dispatcher that init is complete
+        Dispatchers.API.onInitComplete()
```

Comment:
This should probably be at the end of this function, right before the comment for the `ProcessLifecycleOwner.get()...`. Before that point no metric is recorded anyway, they're just delayed in a less controlled way.

Your change here is honestly a *great* improvement for the Glean SDK on Android: we can now control dispatching things instead of relying on the fact that we're using a thread pool with a single consumer. We might even expand that pool, if needed! This wasn't possible before as we relied on the single thread to have more control on the init behaviour and make sure that nothing got recorded before metrics/events pings were generated ;)